source/MaterialXRuntime/RtObject.cpp

```diff
@@ -75,6 +75,28 @@ void RtObject::removeMetadata(const RtToken& name)
     hnd()->asA<PvtObject>()->removeMetadata(name);
 }
 
+size_t RtObject::numMetadata() const
+{
+    return hnd()->asA<PvtObject>()->getMetadataOrder().size();
+}
+
+const RtToken& RtObject::getMetadataName(size_t index) const
+{
+    return hnd()->asA<PvtObject>()->getMetadataOrder()[index];
+}
+
+RtTypedValue* RtObject::getMetadata(size_t index)
```

Comment:
I guess it's complete but do you really need index and name based methods to get values as you can use the name to access the data already?